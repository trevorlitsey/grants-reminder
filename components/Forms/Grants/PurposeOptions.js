import { object, string, func } from 'prop-types';

const PurposeOptions = (props) => {

	const { purposes, selected, handleChange } = props;

	return (
		<select className="u-full-width two columns" name="purpose" value={selected} onChange={handleChange}>
			<option>Purpose</option>
			{Object.keys(purposes).map(key => {
				const purpose = purposes[key];
				return <option key={key} value={purpose} data-test="purpose-option">{purpose}</option>
			})}
		</select>
	)
}

PurposeOptions.propTypes = {
	purposes: object.isRequired,
	selected: string,
	handleChange: func.isRequired,
}

export default PurposeOptions;