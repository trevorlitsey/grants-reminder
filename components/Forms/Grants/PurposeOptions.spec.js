import React from 'react';
import { shallow } from 'enzyme';

import PurposeOptions from './PurposeOptions';

describe('PurposeOptions', () => {

	it('should render without crashing', () => {
		const wrapper = shallowRenderPurposeOptions();
		expect(wrapper).toMatchSnapshot();
	})

	it('should render correct number of purpose options from props', () => {
		const purposes = {
			one: 'education',
			two: 'capital campaign',
		}
		const wrapper = shallowRenderPurposeOptions({ purposes });
		expect(wrapper.find('[data-test="purpose-option"]').length).toEqual(Object.keys(purposes).length);
	})

})

function shallowRenderPurposeOptions(props = {}) {
	const propsToUser = {
		purposes: {
			one: 'education',
			two: 'capital campaign',
			three: 'development',
		},
		selected: 'development',
		handleChange: () => { },
		...props,
	}
	return shallow(<PurposeOptions {...propsToUser} />)
}