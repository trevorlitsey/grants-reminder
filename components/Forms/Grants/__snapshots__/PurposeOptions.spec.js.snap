// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurposeOptions should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PurposeOptions
    handleChange={[Function]}
    purposes={
      Object {
        "one": "education",
        "three": "development",
        "two": "capital campaign",
      }
    }
    selected="development"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
          Purpose
        </option>,
        Array [
          <option
            data-test="purpose-option"
            value="education"
          >
            education
          </option>,
          <option
            data-test="purpose-option"
            value="capital campaign"
          >
            capital campaign
          </option>,
          <option
            data-test="purpose-option"
            value="development"
          >
            development
          </option>,
        ],
      ],
      "className": "u-full-width two columns",
      "name": "purpose",
      "onChange": [Function],
      "value": "development",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Purpose",
        },
        "ref": null,
        "rendered": "Purpose",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "one",
        "nodeType": "host",
        "props": Object {
          "children": "education",
          "data-test": "purpose-option",
          "value": "education",
        },
        "ref": null,
        "rendered": "education",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "two",
        "nodeType": "host",
        "props": Object {
          "children": "capital campaign",
          "data-test": "purpose-option",
          "value": "capital campaign",
        },
        "ref": null,
        "rendered": "capital campaign",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "three",
        "nodeType": "host",
        "props": Object {
          "children": "development",
          "data-test": "purpose-option",
          "value": "development",
        },
        "ref": null,
        "rendered": "development",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Purpose
          </option>,
          Array [
            <option
              data-test="purpose-option"
              value="education"
            >
              education
            </option>,
            <option
              data-test="purpose-option"
              value="capital campaign"
            >
              capital campaign
            </option>,
            <option
              data-test="purpose-option"
              value="development"
            >
              development
            </option>,
          ],
        ],
        "className": "u-full-width two columns",
        "name": "purpose",
        "onChange": [Function],
        "value": "development",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Purpose",
          },
          "ref": null,
          "rendered": "Purpose",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "one",
          "nodeType": "host",
          "props": Object {
            "children": "education",
            "data-test": "purpose-option",
            "value": "education",
          },
          "ref": null,
          "rendered": "education",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "two",
          "nodeType": "host",
          "props": Object {
            "children": "capital campaign",
            "data-test": "purpose-option",
            "value": "capital campaign",
          },
          "ref": null,
          "rendered": "capital campaign",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "three",
          "nodeType": "host",
          "props": Object {
            "children": "development",
            "data-test": "purpose-option",
            "value": "development",
          },
          "ref": null,
          "rendered": "development",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
